build with docker failing

this is not the fixed-yesterday debian-8-container issue.
this is the new debian-9 container failing.
i am super-unfamiliar with docker, so i might be getting some of the terms wrong... 

it seems to pull/download the same docker fragments twice. 
then runs out of space. (afaict _not_ in the buildvm)


build log, with some boring middle sections stripped:
------------------------------------------------------------------------
Sending build context to Docker daemon 121.9 kB
Step 1/9 : FROM ocaml/opam2@sha256:feebac4b6f9df9ed52ca1fe7266335cb9fdfffbdc0f6ba4f5e8603ece7e8b096
Trying to pull repository docker.io/ocaml/opam2 ... 
sha256:feebac4b6f9df9ed52ca1fe7266335cb9fdfffbdc0f6ba4f5e8603ece7e8b096: Pulling from docker.io/ocaml/opam2
bc9ab73e5b14: Pulling fs layer
26d9a6276145: Pulling fs layer
[...]
601d12a29a64: Verifying Checksum
601d12a29a64: Download complete
Trying to pull repository registry.fedoraproject.org/ocaml/opam2 ... 
Trying to pull repository quay.io/ocaml/opam2 ... 
Trying to pull repository registry.access.redhat.com/ocaml/opam2 ... 
Trying to pull repository registry.centos.org/ocaml/opam2 ... 
Trying to pull repository docker.io/ocaml/opam2 ... 
sha256:feebac4b6f9df9ed52ca1fe7266335cb9fdfffbdc0f6ba4f5e8603ece7e8b096: Pulling from docker.io/ocaml/opam2
bc9ab73e5b14: Pulling fs layer
26d9a6276145: Pulling fs layer
[...]
601d12a29a64: Verifying Checksum
601d12a29a64: Download complete
failed to register layer: Error processing tar file(exit status 1): write /home/opam/.opam/4.04/bin/ocamldoc.opt: no space left on device

losing state is actualy semi-intentional there, my usual qubes-mirage-build wouldnt use the docker path either, but is broken right now for some xen-pkg related reasons.

anyways, build with docker some more, now (with symlink) it doesnt run out of space anymore, builds stuff a lot. then fails hard near what looks like endgame linking:

--------------
File "unikernel.ml", line 1:
Error: The files /home/opam/.opam/4.04/lib/mirage-qubes/qubes.cmi
       and uplink.cmi make inconsistent assumptions over interface Qubes
Command exited with code 2.
run ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-tags'
     'predicate(mirage_xen),warn(A-4-41-42-44),debug,bin_annot,strict_sequence,principal,safe_string'
     '-pkgs'
     'cstruct,functoria-runtime,lwt,mirage-clock-freestanding,mirage-logs,mirage-nat,mirage-net-xen,mirage-qubes,mirage-runtime,mirage-types,mirage-types-lwt,mirage-xen,netchannel,shared-memory-ring,tcpip,tcpip.arpv4,tcpip.stack-direct,tcpip.xen,vchan'
     '-cflags' '-g' '-lflags'
     '-g,-dontlink,unix,-dontlink,str,-dontlink,num,-dontlink,threads'
     '-tag-line' '<static*.*>: warn(-32-34)' '-X' '_build-solo5-hvt' '-X'
     '_build-ukvm' 'main.native.o']: exited with 10
Makefile:18: recipe for target 'build' failed
make: *** [build] Error 1

--- a/README.md
+++ b/README.md
@@ -16,13 +16,16 @@ Pre-built binaries are available from the [releases page][].
 
 Clone this Git repository and run the `build-with-docker.sh` script:
 
-    sudo yum install docker
+    sudo ln -s /var/lib/docker /home/user/docker
+    sudo dnf install docker
     sudo systemctl start docker
     git clone https://github.com/talex5/qubes-mirage-firewall.git
     cd qubes-mirage-firewall
     sudo ./build-with-docker.sh
 
 This took about 10 minutes on my laptop (it will be much quicker if you run it again).
+The symlink step at the start isn't needed if your build VM is standalone.
+It gives Docker more disk space and avoids losing the Docker image cache when you reboot the Qube.
 
 You can also build without Docker, as for any normal Mirage unikernel;
 see [the Mirage installation instructions](https://mirage.io/wiki/install) for details.

